core/thread/credentials.go

```diff
@@ -0,0 +1,98 @@
+package thread
+
+import (
+	"github.com/libp2p/go-libp2p-core/crypto"
+)
+
+// Signature of an ID
+type Signature []byte
+
+// Credentials are used to prove a caller identity, which is needed
+// for thread ACL checks.
+type Credentials interface {
+	// ThreadID must return the signed thread ID.
+	ThreadID() ID
+
+	// Sign must return the public key and signature which can
+	// be used to verify the caller.
+	Sign() (crypto.PubKey, Signature, error)
+}
+
+// DefaultCreds are useful for either threads that don't require
+// credentials or when the private key is available for local signing.
+type DefaultCreds struct {
+	threadID ID
+	privKey  crypto.PrivKey
+}
+
+// NewDefaultCreds returns new default credentials.
+func NewDefaultCreds(id ID, opts ...CredOption) Credentials {
+	args := &CredOptions{}
+	for _, opt := range opts {
+		opt(args)
+	}
+	return DefaultCreds{threadID: id, privKey: args.PrivKey}
+}
+
+func (c DefaultCreds) ThreadID() ID {
+	return c.threadID
+}
+
+func (c DefaultCreds) Sign() (crypto.PubKey, Signature, error) {
```

Comment:
The default way to use credentials comes with a signing method that used the provided private key. Note, the caller does not have access to the key.